src/main/java/gov/cabinetofice/gapuserservice/service/DepartmentService.java

```diff
@@ -20,4 +23,28 @@ public List<DepartmentDto> getAllDepartments() {
                 .map(departmentMapper::departmentToDepartmentDto)
                 .collect(Collectors.toList());
     }
+
+    public Optional<Department> getDepartmentById(int id) {
+        return departmentRepository.findById(id);
+    }
+
+    public Department updateDepartment(Department department, String departmentName, String ggisId) throws DepartmentNotFoundException {
+         department.setName(departmentName);
+         department.setGgisID(ggisId);
+        departmentRepository.save(department);
+        return department;
+    }
+
+    public void deleteDepartment(int id) {
+        Department department = departmentRepository.findById(id).orElseThrow(() -> new DepartmentNotFoundException("Department not found"));
+        departmentRepository.delete(department);
```

Comment:
JPA lets you define delete queries where you can say something like "deleteById", this would be marginally more performant